src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -283,17 +324,32 @@ def window(
                 copy of the data.
 
         Raises:
-            IndexError: When requesting a window with invalid timestamps.
+            IndexError: When start and end are not both datetime or index.
 
         Returns:
             The requested window
         """
-        if start > end:
+        if self.count_covered == 0:
+            return np.array([]) if isinstance(self._buffer, np.ndarray) else []
+
+        # If both are indices or None convert to datetime
+        if not isinstance(start, datetime) and not isinstance(end, datetime):
+            start, end = self._index_to_slice(start, end)
+            start = self.index_to_datetime(start)
+            end = self.index_to_datetime(end)
+
+        # Here we should have both as datetime
+        if not (isinstance(start, datetime) and isinstance(end, datetime)):
```

Comment:
What if `end` is not a `datetime` but `start` is? Does it really works in that case or should this be
```suggestion
        if not (isinstance(start, datetime) or not isinstance(end, datetime)):
```
Instead?

Reply:
In the current case it only works if both are same type, if not, this one would throw. The idea was that either both are integer or both are datetime. Technically we could also support a mix of both, I was not sure whether we should. What do you think?

Reply:
I would support the mix, unless there is a very strong case for it. My point above is, unless I'm understanding the code wrongly, the `if` above will not raise an error (at least not explicitly) if `end` is not a `datetime` but `start` is, right?

You have:

```py
        # If both are indices or None convert to datetime
        if not isinstance(start, datetime) and not isinstance(end, datetime):
            start, end = self._to_covered_indices(start, end)
            start = self.get_timestamp(start)
            end = self.get_timestamp(end)

        # Here we should have both as datetime
        if not (isinstance(start, datetime) and isinstance(end, datetime)):
            raise IndexError(...)
```

Here you continue operating as if both `start` and `end` were `datetime`s,
So you should probably raise if **any** if `start` **or** `end` is **not** a `datetime` above.

Reply:
Oh, sorry, I see I misread the parenthesis :facepalm: 

So scratch all the above and take the suggestion just as a code readability improvement :laughing:

Reply:
Updated according to your suggestion. 

 > I would support the mix, unless there is a very strong case for it.
 
Could you clarify on this please, would you support a mix of datetime and int indices (e.g. `moving_window[datetime(2023, 9, 25): -5]`) or not? The current version does not, and honestly I find it confusing to use. We can still support it later if we find good use-cases.

Reply:
Sorry, that was a *typo*, I meant "I would **not** support" :disappointed:

Reply:
Ok, that's what's currently implemented so this PR is ready. @llucax